### PR TITLE
Removes request for restarting the editor after resizing the main font 

### LINKED ISSUE #55942: Main font size editor setting incorrectly requires restart
### Godot version

f09d884

### System information

W10

### Issue description

When you change "Main Font Size" in Editor Settings, you get that yellow text saying changes won't be applied until restart. But this is a lie.
btCwwoGoiI
This might be not the only setting with this problem.

### Steps to reproduce

See GIF.

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by volokh0x**
Fixes #55942

### CODE REVIEW

**Comment by pycbouh** [comment]
I don't think we should do this, and I second Akien's argument in the original issue. We don't have consistent respect for the theme changes in the editor. It *should* be updated on fly, but in practice a lot of contributors over the years didn't handle this situation properly and it's unreliable to assure the user that the you won't have any issues if you change theming settings without a restart.

**Comment by Calinou** [comment]
I think it makes sense to remove this warning, but only after the caveats have been documented by merging https://github.com/godotengine/godot/pull/48548.

**Comment by KoBeWi** [comment]
>  in practice a lot of contributors over the years didn't handle this situation properly and it's unreliable to assure the user that the you won't have any issues

If such cases still exist, they will be spotted rather quickly and get fixed. It's not a big problem; depends on whether we want the editor to reliably reload the theme or not. Lots of work was done towards this and it will go to waste if we just assume that the editor might still break in *some* places.

**Comment by pycbouh** [comment]
> If such cases still exist, they will be spotted rather quickly and get fixed.

Well, after fixing a lot of theming propagation in the editor I have to say, it doesn't get spotted quickly, even when problems should be obvious to the user. But yes, intentionally allowing the editor to bug out for the sake of highlighting problems would work, at the cost of user experience.

**Comment by Norrox** [comment]
i would say we need this button on the inspector setting page aswell :P the editor need to be reloaded when changing for example this property
Skärmbild 2021-12-30 130103

**Comment by pycbouh** [comment]
> i would say we need this button on the inspector setting page aswell :P the editor need to be reloaded when changing for example this property

Not really related to this PR, but I don't think there is any reason for a restart there. But you probably need to re-open your object in the inspector to see the changes.

**Comment by Norrox** [comment]
> > i would say we need this button on the inspector setting page aswell :P the editor need to be reloaded when changing for example this property
> 
> Not really related to this PR, but I don't think there is any reason for a restart there. But you probably need to re-open your object in the inspector to see the changes.

Yeah your right, it works in master but totally broken in 3.4.x i guess i need to make an issue :)

**Review by Calinou — APPROVED** [review]
I agree supporting run-time font size changes is something we should aim to support as much as possible, in the interest of accessibility. Therefore, it makes sense to merge this after this is rebased.

File: editor/editor_settings.cpp
Comment (akien-mga) [inline editor/editor_settings.cpp:407]: You can also remove `PROPERTY_USAGE_DEFAULT` since that's the default.

**Pull request closed without merging**
